vowpalwabbit/core/src/reductions/metrics.cc

```diff
@@ -137,38 +137,40 @@ void predict_or_learn(metrics_data& data, T& base, E& ec)
 
 void VW::reductions::output_metrics(VW::workspace& all)
 {
-  if (all.options->was_supplied("extra_metrics"))
+  metrics_manager& manager = all.global_metrics;
+  if (manager.are_metrics_enabled())
   {
-    std::string filename = all.options->get_typed_option<std::string>("extra_metrics").value();
-    VW::metric_sink list_metrics;
-
-    all.l->persist_metrics(list_metrics);
-
-    for (auto& metric_hook : all.metric_output_hooks) { metric_hook(list_metrics); }
-
-    list_metrics.set_uint("total_log_calls", all.logger.get_log_count());
+    auto additional_metrics = [&all](metric_sink& sink) -> void
```

Comment:
I don't think we should add this extra stuff as a callback. That means if you were to call output_metrics twice it would add the callback twice.

I think we should leave the callbacks as is and add these extra metrics outside of being a callback.

Reply:
I agree this should be decoupled from output_metrics. I think it will still need to be added as a callback, since there is no way to interact with the metrics_sink directly in the new design. I added a callback in parse_args so it will not be called multiple times (eg in `output_metrics` or `list_to_json_file`)